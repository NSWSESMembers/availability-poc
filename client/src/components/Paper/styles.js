import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    elevation: 1,
    borderRadius: 2,
    backgroundColor: 'white',
    paddingTop: 10,
    paddingBottom: 10,
    paddingLeft: 8,
    paddingRight: 8,
    marginLeft: 10,
    marginRight: 10,
    marginTop: 6,
    marginBottom: 0,
  },
  textContainer: {
    flexDirection: 'column',
    flexShrink: 1,
  },
  text: {
    textAlignVertical: 'bottom',
    includeFontPadding: false,
    flex: 0,
    fontSize: 12,
    color: '#6B6B6B',
  },
  titleText: {
    textAlignVertical: 'top',
    includeFontPadding: false,
    flex: 0,
    marginRight: 8,
    fontWeight: 'bold',
  },
});

export default styles;
