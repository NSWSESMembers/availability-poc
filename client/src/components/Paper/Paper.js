import React from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';

import styles from './styles';

const Paper = ({ title, text }) => {
  const containerStyles = [styles.container];

  return (
    <View>
      <View style={containerStyles}>
        <View style={styles.textContainer}>
          {title && <Text style={styles.titleText}>{title}</Text>}
          {text && <Text style={styles.text}>{text}</Text>}
        </View>
      </View>
    </View>
  );
};

Paper.propTypes = {
  title: PropTypes.string,
  text: PropTypes.string,
};

export default Paper;
