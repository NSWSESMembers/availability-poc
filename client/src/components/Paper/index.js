import Paper from './Paper';
import styles from './styles';

export { Paper, styles };
