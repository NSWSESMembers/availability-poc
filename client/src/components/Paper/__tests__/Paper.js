import React from 'react';
import renderer from 'react-test-renderer';

import Paper from '../Paper';

test('renders correctly', () => {
  const tree1 = renderer.create(
    <Paper
      title="title"
      text="text"
    />,
  ).toJSON();
  expect(tree1).toMatchSnapshot();
});
