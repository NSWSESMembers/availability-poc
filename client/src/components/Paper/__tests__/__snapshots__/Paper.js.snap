// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderRadius": 2,
          "elevation": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 0,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 6,
          "paddingBottom": 10,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 10,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "column",
          "flexShrink": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 0,
            "fontWeight": "bold",
            "includeFontPadding": false,
            "marginRight": 8,
            "textAlignVertical": "top",
          }
        }
      >
        title
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#6B6B6B",
            "flex": 0,
            "fontSize": 12,
            "includeFontPadding": false,
            "textAlignVertical": "bottom",
          }
        }
      >
        text
      </Text>
    </View>
  </View>
</View>
`;
